src/main/java/com/carrotcreative/cream/util/HashingUtil.java

```diff
@@ -0,0 +1,52 @@
+package com.carrotcreative.cream.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import android.support.annotation.NonNull;
+
+public class HashingUtil
+{
+    private static final int CACHE_SIZE = 50;
```

Comment:
50 is the number of entries in the LruCache. The default implementation is the number of entries and if we override the  sizeOf(K, V) then its in terms of memory.

```
public LruCache (int maxSize)

Parameters
maxSize	for caches that do not override sizeOf(K, V), this is the maximum number of entries 
in the cache. 
For all other caches, this is the maximum sum of the sizes of the entries in this cache. 
```

Ref: http://developer.android.com/reference/android/util/LruCache.html#LruCache(int)